OsmAnd/src/net/osmand/plus/activities/actions/ShareLocation.java

```diff
@@ -44,16 +44,16 @@ public void onClick(DialogInterface dialog, int which) {
 				final int zoom = args.getInt(MapActivityActions.KEY_ZOOM);
 				try {
 					final String shortOsmUrl = MapUtils.buildShortOsmUrl(latitude, longitude, zoom);
-					final String appLink = "http://download.osmand.net/go?lat=" + ((float) latitude) + "&lon=" + ((float) longitude) + "&z=" + zoom;
-					String sms = mapActivity.getString(R.string.send_location_sms_pattern, shortOsmUrl, appLink);
+					final String geoUrl = MapUtils.buildGeoUrl(latitude, longitude, zoom);
+					String sms = mapActivity.getString(R.string.send_location_sms_pattern, geoUrl, shortOsmUrl);
 					if (which == 0) {
-						sendEmail(shortOsmUrl, appLink);
```

Comment:
I tried to figure out what that `appLink` (aka download.osmand.net/go?lat=12.6313&lon=-7.9955&z=8) did, and the only thing I could get it to do was redirect to Google Maps.  Right now, this pull request sends this as the message:

```
Location: geo:48.17396,16.365845?z=12
https://openstreetmap.org/go/0JrID2H--?m
```

So it is the same as the old message, I just replaced the download.osmand.net link with a direct, secure link to OpenStreetMap.